docs/design/proposals/healer-controller.md

```diff
@@ -0,0 +1,43 @@
+# Introduction
+There is [external health monitoring](https://github.com/kubernetes-csi/external-health-monitor/) which does mostly similar stuff, but actions are not in the scope of it, all it can do is log the health condition of the mounts.
+
+# Volume Healer Controller
+
+**Now the obvious question we expect is "what do we achieve with the new controller?"**
+* Achieve a way to monitor the health of volumes.
+* Ceph-CSI has healer component which take cares of restarting the userspace processes when the node plugin is restarted, but in many places it uses Kubernetes specific code, with this controller model we are moving out the Kubernetes specific code out, and help make Ceph-CSI orchestrator independent.
```

Comment:
cephcsi name is not required here as its a general component.

Reply:
I'm convinced with a comment by Niels earlier:

```
I think you are concerned that the CSI-driver container in the pod restarted, and not the pod was re-created.
Only in that case, the CSIAddonsNode object would exist. It is not clear to me if this is something that can happen in reality.
``